src/java/org/apache/cassandra/net/OutboundTcpConnection.java

```diff
@@ -236,8 +238,10 @@ else if (socket != null || connect())
                     else
                     {
                         // clear out the queue, else gossip messages back up.
-                        drainedMessages.clear();
-                        backlog.clear();
+                        int cleared = clearQueueWithFailureCallback(i, drainedMessages, drainedMessageSize, backlog);
```

Comment:
What specific callback behavior are you thinking will help here? Or is this just for visibility (eg `failuresUpdater.incrementAndGet(this)`)

Reply:
if you invoke the failure callback, I believe AbstractWriteResponseHandler will be able to tell whether the latest failure means we won't get the required number of responses, then call signal() so we're no longer blocking in StorageProxy

Reply:
the unit test shows this. the get() returns faster than the write timeout w/ this change